### PR TITLE
Address various review comments for Darwin OTA delegate bridge.

### LINKED ISSUE #20763: Address review issues in Darwin OTAProviderDelegateBridge
#### Problem
See https://github.com/project-chip/connectedhomeip/pull/20516#discussion_r921545649 and https://github.com/project-chip/connectedhomeip/pull/20516#pullrequestreview-1039429790

 #### Proposed Solution
Fix those issues.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
Fixes https://github.com/project-chip/connectedhomeip/issues/20763

#### Problem
Review comments that need addressing.

#### Change overview
Address them.

#### Testing
No behavior changes except in edge cases with integer overflow.

### CODE REVIEW

File: src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm [hunk 1]
```diff
@@ -50,7 +53,12 @@
     if ([strongDelegate respondsToSelector:@selector(handleQueryImage:completionHandler:)]) {
         if (strongDelegate && mQueue) {
             auto * commandParams = [[MTROtaSoftwareUpdateProviderClusterQueryImageParams alloc] init];
-            ConvertToQueryImageParams(commandData, commandParams);
+            CHIP_ERROR err = ConvertToQueryImageParams(commandData, commandParams);
+            if (err != CHIP_NO_ERROR) {
+                commandObj->AddStatus(commandPath, chip::Protocols::InteractionModel::Status::InvalidCommand);
+                handle.Release();
```
Comment (carol-apple) [inline src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm:59]: Now I wonder if we should move line 48-49 to below this block so we don't need to make an extra `handle.Release` call here. The handler and command path really only needs to be cached before the dispatch below..

File: src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm [hunk 2]
```diff
@@ -50,7 +53,12 @@
     if ([strongDelegate respondsToSelector:@selector(handleQueryImage:completionHandler:)]) {
```
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm:53]: Good catch, no.

File: src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm (hunk 1 above)
Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm:59]: We could probably do that, yes.

**Review by carol-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
Fast-tracking platform change.

**Pull request merged**